src/renderer/kube-object.store.ts

```diff
@@ -195,29 +194,32 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
   }
 
   // collect items from watch-api events to avoid UI blowing up with huge streams of data
-  protected eventsBuffer = observable<IKubeWatchEvent<KubeJsonApiData>>([], { deep: false });
+  protected eventsBuffer = observable.array<IKubeWatchEvent<KubeJsonApiData>>([], { deep: false });
 
   protected bindWatchEventsUpdater(delay = 1000) {
-    return reaction(() => this.eventsBuffer.toJS()[0], this.updateFromEventsBuffer, {
+    kubeWatchApi.onMessage.addListener(({ store, data }: IKubeWatchMessage<T>) => {
+      if (!this.isLoaded || store !== this) return;
+      this.eventsBuffer.push(data);
+    });
+
+    reaction(() => this.eventsBuffer.length > 0, this.updateFromEventsBuffer, {
       delay
     });
   }
 
-  subscribe(apis = [this.api]) {
-    return KubeApi.watchAll(...apis);
+  getSubscribeApis(): KubeApi[] {
```

Comment:
Could you explain why `getSubscribeApis()` is introduced in this PR? Is it more convenient way to override subscribed apis per store?

Reply:
Yes, it better to have "in place" since some stores might subscribe for multiple apis, e.g. `RolesStore`, `RoleBindingsStore`.
Previously handled in `subscribe()` but having apis separately gives more freedom of using this (initially refactored to  be used in `KubeWatchApi`).